src/components/data/HangingGarden/__stories__/WorkOrderGarden/utils/procosys.ts

```diff
@@ -0,0 +1,221 @@
+    [proCoSysStatuses.Cancelled]: 0xf4937b,
+    [proCoSysStatuses.SentDC]: 0x70a5ff,
+    [proCoSysStatuses.ToMC]: 0xf1beb1,
+    [proCoSysStatuses.Prepared]: 0xb8b8b8,
+    [proCoSysStatuses.NoStatus]: 0xec3e13,
+};
+
+export const proCoSysWorkOrderProgressColorMap: { [hex: string]: number } = {
+    [proCoSysStatuses.ASBuiltCompleted]: 0x99beff,
+    [proCoSysStatuses.SentToPlanning]: 0x001c4d,
+    [proCoSysStatuses.MCDocsPrepared]: 0xecec13,
+    [proCoSysStatuses.ToField]: 0x22c32a,
+    [proCoSysStatuses.FromField]: 0x52e059,
+    [proCoSysStatuses.ComplByMC]: 0x004bcc,
+    [proCoSysStatuses.Cancelled]: 0xf06542,
+    [proCoSysStatuses.SentDC]: 0x337eff,
+    [proCoSysStatuses.ToMC]: 0xe8927d,
+    [proCoSysStatuses.Prepared]: 0x999999,
+    [proCoSysStatuses.NoStatus]: 0xbd320f,
+};
+
+const prepareMaterialStatus = (status: string): string[] => {
+    status = status.toLowerCase();
+
+    let number = status.replace(/[^0-9]+/, '');
+    if (number.length === 1) {
+        number = '0' + number;
+    }
+
+    if (!number.length) {
+        return [status];
+    }
+
+    return [status, 'm' + number];
+};
+
+const woHasMaterialStatus = (workOrder: WorkOrderType, ...statuses: string[]) => {
+    const materialStatuses = statuses
+        .map((status) => prepareMaterialStatus(status))
+        .reduce((all, current) => all.concat(current), []);
+    const woMaterialStatus = workOrder.materialStatus.toLowerCase();
+    return materialStatuses.filter(
+        (materialStatus) => woMaterialStatus.indexOf(materialStatus) === 0
+    ).length;
+};
+
+export const orderedProCoSysStatuses: string[] = [
+    proCoSysStatuses.NoStatus,
+    proCoSysStatuses.Prepared,
+    proCoSysStatuses.ToMC,
+    proCoSysStatuses.MCDocsPrepared,
+    proCoSysStatuses.ToField,
+    proCoSysStatuses.FromField,
+    proCoSysStatuses.ComplByMC,
+    proCoSysStatuses.Cancelled,
+    proCoSysStatuses.SentDC,
+    proCoSysStatuses.ASBuiltCompleted,
+    proCoSysStatuses.SentToPlanning,
+];
+
+export const materialOk = (workOrder: WorkOrderType) =>
+    woHasMaterialStatus(workOrder, 'm12', 'm13', 'mn');
+export const materialAvailable = (workOrder: WorkOrderType) =>
+    woHasMaterialStatus(workOrder, 'm7', 'm9', 'm10', 'm11', 'mn');
+export const materialNotAvailable = (workOrder: WorkOrderType) =>
+    !workOrder.materialStatus || woHasMaterialStatus(workOrder, 'NS', 'm2', 'm3', 'm4', 'm5', 'm6');
+export const getFollowUpStatus = (workOrder: WorkOrderType) => {
+    const status = getWoStatus(workOrder);
+    const statusIndex = orderedProCoSysStatuses.indexOf(status);
+
+    if (workOrder.projectProgress.toString() === '100') {
+        return followUpStatuses.WOFinished;
+    } else if (materialOk(workOrder) && [4, 5, 6, 7, 8, 9, 10].indexOf(statusIndex) > -1) {
+        return followUpStatuses.MaterialAndWoOk;
+    } else if (
+        materialAvailable(workOrder) &&
+        [3, 4, 5, 6, 7, 8, 9, 10].indexOf(statusIndex) > -1
```

Comment:
use enum or a function.
what if somebody changes the order of this array?
2nd it really hard to read and understand, what is status of status codes index X?